pennylane/templates/tensornetworks/mps.py

```diff
@@ -0,0 +1,177 @@
+            )
+        else:
+            self.n_blocks = _n_blocks
+
+        allowed_templates = [qml.templates.BasicEntanglerLayers, qml.templates.StronglyEntanglingLayers] #, qml.templates.layers.SimplifiedTwoDesign]
+
+        if block not in allowed_templates: 
+            raise AssertionError("The input block is not included yet in the allowed templates block. Try use MPS_from_function instead.")
+
+        self.block = block
+
+        if block_arg == None:
+
+            self.block_arg = {}
+
+        else:
+            self.block_arg = block_arg
+            self.block_specified()
+
+
+        self.ind_gates = compute_indices_MPS(n_wires, loc)
+
+        super().__init__(weights, wires=wires, do_queue=do_queue, id=id)
+
+    def block_specified(self):
+
+        # An naive implementation to further specify the template block
+        # Non-essential for now, a function to further specified the function
+
+        if self.block == qml.templates.BasicEntanglerLayers and "rotation" in self.block_arg.keys():
+
+            self.block =lambda weights, wires: qml.templates.BasicEntanglerLayers(weights, wires, rotation=self.block_arg["rotation"])
+
+        if self.block == qml.templates.StronglyEntanglingLayers and  "ranges" in self.block_arg.keys():
+
+            self.block = lambda weights, wires: qml.templates.StronglyEntanglingLayers(weights, wires, ranges=self.block_arg["ranges"]) 
+
+        #if self.block == qml.templates.layers.SimplifiedTwoDesign and "initial_layer_weights" in self.block_arg.keys():
+
+        #    self.block = lambda weights, wires: qml.templates.SimplifiedTwoDesign(initial_layer_weights=self.block_arg["initial_layer_weights"], weights, wires)
+
+        return
+
+        
+    def expand(self):
+        with qml.tape.QuantumTape() as tape:
+            for idx, w in enumerate(self.ind_gates):
+                self.block(weights=self.weights[...,idx], wires=w.tolist()) # MEASURE THE LAST QUBIT
+        return tape
+
+
+    @staticmethod
+    def shape(n_wires, block_shape):
+
+        r"""Returns the expected shape of the weights tensor.
+        Args:
+            n_wires (int): total number of wires
+            block_shape (tuple[int]): shape of the block circuit in (n_layers [int], loc [int], n_params [int])
+        Returns:
+            tuple[int]: shape
+        """
+
+        loc = block_shape[1]
+
+        #assert n_wires%2 == 0, "n_wires should be an even integer"
+        #assert loc%2 == 0 and loc != 0, "loc should be an even integer"
+        #assert loc <= n_wires, "loc should be smaller or equal than num_wires"
+
+        n_blocks = int(n_wires/(loc/2)-1)
+
+        return block_shape + (n_blocks,)
+
+class MPS_from_function(Operation):
+    
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+    
+    def __init__(self, wires, loc, block, n_params_block, weights=None, ranges=None, imprimitive=None, do_queue=True, id=None):
```

Comment:
```suggestion
    def __init__(self, wires, loc, block, n_params_block, weights=None, do_queue=True, id=None):
```
I removed `imprimitive` and `ranges` since they are not used in the template.
I think there should be a way to calculate `n_params_block` inside the template. One way to hack this would be to build a tape and run `tape.trainable_params`. But I think there might be a better way